datastore/datastore/storagebackend/postgresql/postgresql.go

```diff
@@ -21,6 +22,82 @@ func (sbe *PostgreSQL) Description() string {
 	return "PostgreSQL database"
 }
 
+// setTSUniqueMainCols extracts into tsStringMdataPBNamesUnique the columns comprising constraint
+// unique_main in table time_series.
+//
+// Returns nil upon success, otherwise error.
+func (sbe *PostgreSQL) setTSUniqueMainCols() error {
+
+	query := `
+		SELECT pg_get_constraintdef(c.oid)
+		FROM pg_constraint c
+		JOIN pg_namespace n ON n.oid = c.connamespace
+		WHERE conrelid::regclass::text = 'time_series'
+			AND conname = 'unique_main'
+			AND contype = 'u'
+	`
+
+	row := sbe.Db.QueryRow(query)
+
+	var result string
+	err := row.Scan(&result)
+	if err != nil {
+		return fmt.Errorf("row.Scan() failed: %v", err)
+	}
+
+	pattern := `\((.*)\)`
+	re := regexp.MustCompile(pattern)
+	matches := re.FindStringSubmatch(result)
+	if len(matches) != 2 {
+		return fmt.Errorf("'%s' didn't match regexp pattern '%s'", result, pattern)
+	}
+
+	// create tsStringMdataPBNamesUnique
+	tsStringMdataPBNamesUnique = strings.Split(matches[1], ",")
+	for i := 0; i < len(tsStringMdataPBNamesUnique); i++ {
+		tsStringMdataPBNamesUnique[i] = strings.TrimSpace(tsStringMdataPBNamesUnique[i])
+	}
+
+	return nil
+}
+
+// setUpsertTSInsertCmd sets upsertTSInsertCmd to be used by upsertTS.
+func setUpsertTSInsertCmd() {
+
+	cols := getTSMdataCols()
+
+	formats := make([]string, len(cols))
+	for i := 0; i < len(cols); i++ {
+		formats[i] = "$%d"
+	}
+
+	updateExpr := []string{}
+	for _, col := range getTSMdataColsUniqueCompl() {
+		updateExpr = append(updateExpr, fmt.Sprintf("%s = EXCLUDED.%s", col, col))
```

Comment:
We don't care about the order of the different columns here, correct? If I understand Go correctly, the return value of `etTSMdataColsUniqueCompl` will be randomized, correct?

Reply:
Correct! And correct!

Reply:
(I think it's a good thing that golang randomizes map iteration, since it forces developers to treat a map as an unordered structure where you can't rely on some arbitrary system-dependent order)